### PR TITLE
:gift_heart: Migrating to sink flag for --to param

### LINKED ISSUE #160: The `--sender-namespace` should be filled in from `--namespace` option, if given
The default value of `--sender-namespace` option should be equal to the value of a `--namespace`, if `--namespace` option is given. Otherwise, it should be equal to current context namespace.

/kind bug
/kind good-first-issue

### LINKED ISSUE #202: Comply with `containedctx` linter
The [containedctx](https://github.com/sivchari/containedctx) linter is being included within [golangci-lint](https://golangci-lint.run/usage/linters/). We should make the code comply with this linter.

Violations:

```
pkg/k8s/addressresolver.go:39:2: found a struct that contains a context.Context field (containedctx)
	ctx  context.Context
	^
pkg/k8s/kubeclient.go:105:2: found a struct that contains a context.Context field (containedctx)
	ctx       context.Context
	^
pkg/tests/fakeclients.go:33:2: found a struct that contains a context.Context field (containedctx)
	ctx       context.Context
	^
```

### LINKED ISSUE #282: Align the `--to` param to comply with standard kn `--sink` param
During the development of the kn event plugin, two options for sending have been implemented: `--to` and `--to-url`. Both of them are being described in this doc, in appropriate sections. 

It makes sense to align both of them in single `--to` option, as the sink command line spec format is way easier to write than the current subject-based one. Maybe, to keep the functionality of matching resources by labels and targeting resources besides Knative, but addressable, appropriate contributions can be made to knative client itself to extend the sink spec format to accept subject-style as fallback.

Also, the `--to` will accept http(s) addresses.

/kind enhancement

### LINKED ISSUE #356: The build command should print the event on STDOUT
Currently, the build command prints the event output on the standard error output. This prevents some tools to work properly, while using Unix' pipes or redirections.

The standard output should be used instead.

/kind bug

### PR DESCRIPTION
**Pull request by cardil**
# Changes

- :gift_heart: Migrating to sink flag for --to param
- :broom: Removing the --sender-namespace (the --namespace takes over)
- :broom: Better logging, common with kn

/kind enhancement

# Relations

Fixes #282
Fixes #202
Fixes #356
Fixes #160
Requires https://github.com/knative/client/pull/1968

# Release Notes

```release-note
TBD
```

# Docs

```docs
TBD
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -30,10 +32,14 @@ require (
 	sigs.k8s.io/yaml v1.4.0
 )
 
+// TODO: Remove when https://github.com/knative/client/pull/1968 is merged
+replace knative.dev/client/pkg => github.com/cardil/knative-client/pkg v0.0.0-20240923095307-3a2bcba04752
```
Comment (cardil) [inline go.mod:36]:
```suggestion
```
Remove when https://github.com/knative/client/pull/1968 is merged

Comment (cardil) [reply go.mod:36]: /hold

File: internal/ics/app.go [hunk 2]
```diff
@@ -1,58 +1,52 @@
...
-	return &cobra.Command{
+	c := &cobra.Command{
 		Use:           "ics",
 		SilenceUsage:  true,
 		SilenceErrors: true,
 		RunE:          a.run,
 	}
+	c.SetContext(cli.InitialContext())
+	c.PersistentPreRun = func(cmd *cobra.Command, _ []string) {
+		cli.SetupContext(cmd, zapcore.DebugLevel)
```
Comment (cardil) [inline internal/ics/app.go:31]: On K8s, the sender fails with:

```
mkdir /.cache: permission denied
```

Comment (dsimansk) [reply internal/ics/app.go]: That's addressed in the latest change, right? https://github.com/knative-extensions/kn-plugin-event/pull/368/commits/76a4108e8bb401f336f678b34972b2b35a484583

**Review by dsimansk — APPROVED** [review]
/approve
/lgtm

File: internal/ics/app.go (hunk 2 above)
Comment (cardil) [reply internal/ics/app.go]: Yes, however, I still wonder why e2e tests haven't caught it...

**Comment by cardil** [comment]
@dsimansk Would you mind reviewing https://github.com/knative-extensions/kn-plugin-event/pull/368/commits/da9f541b6692a42034713d660ee0f1192e8c7110, as well?

I added the above as the PR suppose to fix https://github.com/knative-extensions/kn-plugin-event/issues/356

File: go.mod (hunk 1 above)
Comment (cardil) [reply go.mod]: /hold cancel

**Pull request merged**